mods/fire/init.lua

```diff
@@ -3,7 +3,12 @@
 -- Global namespace for functions
 
 fire = {}
-
+fire.features = {
```

Comment:
Here's an example to help see the benefit to it:

A mod wishes to add lava causing rails(they have wooden tie/sleepers) to burn into twisted/burnt rails that lose there function, but return some steel ingots on digging them up.

Currently, if that mod sets groups.flammable on the rail, the default Fire mod will completely remove the rail.  So, the mod has no choice but to create it's own ABM to trigger on rails(which, by default, aren't flammable and normally left untouched by Fire) to transform them.  This means an extra ABM which could be avoided now with the features.on_burn_callback check.

The mod can check that and add it's own custom ABM to burn nodes(that normally aren't) to a burnt version or if the features feature is available on the server (from default game being upgraded), the mod can now instead handle the transform with the callback without needing to add an ABM any more, improving server performance.

I hadn't added to the game_api.txt before it was requested later - I'd included comments to the Fire mod itself for ease-of-access and folder copying into various games.  It's useful to explain in the Fire lua, for those who just want to check out what it does, without looking for a higher-level document that they may not be aware exists.